backends/tfhe-cuda-backend/cuda/src/crypto/ciphertext.cuh

```diff
@@ -39,28 +39,42 @@ __global__ void sample_extract(Torus *lwe_array_out, Torus const *glwe_array_in,
   auto lwe_out = lwe_array_out + input_id * lwe_output_size;
 
   // We assume each GLWE will store the first polynomial_size inputs
-  uint32_t lwe_per_glwe = params::degree;
   auto glwe_in = glwe_array_in + (input_id / lwe_per_glwe) * glwe_input_size;
 
-  // nth is ensured to be in [0, lwe_per_glwe)
-  auto nth = nth_array[input_id] % lwe_per_glwe;
+  // nth is ensured to be in [0, params::degree)
+  auto nth = nth_array[input_id] % params::degree;
 
   sample_extract_mask<Torus, params>(lwe_out, glwe_in, glwe_dimension, nth);
   sample_extract_body<Torus, params>(lwe_out, glwe_in, glwe_dimension, nth);
 }
 
+/**
+ * \brief Extracts LWE ciphertexts from GLWE ciphertexts on the GPU.
+ *
+ * This function extracts LWE ciphertexts from GLWE ciphertexts
+ *
+ * \param stream The CUDA stream to use for the operation.
+ * \param gpu_index The index of the GPU to use.
+ * \param lwe_array_out The output array for the extracted LWE ciphertexts.
+ * \param glwe_array_in The input array of GLWE ciphertexts.
+ * \param nth_array The array of indices specifying which LWE ciphertexts to
+ * extract from each GLWE ciphertext. \param num_nths The number of indices in
+ * the nth_array. \param lwe_per_glwe The number of LWE ciphertexts to extract
```

Comment:
I don't understand why we need lwe_per_glwe, num_nths does not provide the same information?

Reply:
`num_nths` will tell you how many LWEs will be extracted in total. `lwe_per_glwe` will tell you how many LWEs will be extracted from a single GLWE.

Reply:
Ah ok makes sense :+1: :pray: